docs/streams/developer-guide/config-streams.html

```diff
@@ -684,70 +728,64 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
               <td>Consumer</td>
               <td>1000</td>
             </tr>
-            <tr class="row-odd"><td>rocksdb.config.setter</td>
-              <td>Consumer</td>
-              <td>&nbsp;</td>
-            </tr>
             </tbody>
           </table>
         </div>
+        <div class="section" id="parameters-controlled-by-kafka-streams">
+          <h3><a class="toc-backref" href="#id26">Required configuration parameters</a><a class="headerlink" href="#parameters-controlled-by-kafka-streams" title="Permalink to this headline"></a></h3>
+          <p>Kafka Streams assigns the following configuration parameters. If you try to change
+            these settings, your values are ignored and setting them has no effect in a
+            Kafka Streams application.
+          <p>Kafka Streams uses the <code class="docutils literal"><span class="pre">client.id</span></code>
+            parameter to compute derived client IDs for internal clients. If you don't set
+            <code class="docutils literal"><span class="pre">client.id</span></code>, Kafka Streams sets it to
+            <code class="docutils literal"><span class="pre">&lt;application.id&gt;-&lt;random-UUID&gt;</span></code>.
+            <table border="1" class="non-scrolling-table docutils">
+              <colgroup>
+              <col width="50%">
+              <col width="19%">
+              <col width="31%">
+              </colgroup>
+              <thead valign="bottom">
+              <tr class="row-odd"><th class="head">Parameter Name</th>
+              <th class="head">Corresponding Client</th>
+              <th class="head">Streams Default</th>
+              </tr>
+              </thead>
+              <tbody valign="top">
+              <tr class="row-even"><td>auto.create.topics.enable</td>
+              <td>Consumer</td>
+              <td>false</td>
+              </tr>
+              <tr class="row-even"><td>auto.offset.reset</td>
+              <td>Consumer</td>
+              <td>earliest</td>
```

Comment:
Users can overwrite `auto.offset.reset` -- it's just a different default that for the plain `KafkaConsumer` -- same for the following 4 configs.

Reply:
I reworked this a bit, is it accurate?